applications/tari_merge_mining_proxy/src/proxy.rs

```diff
@@ -341,12 +341,15 @@ impl InnerService {
                             status,
                             details: "failed to submit block".to_string(),
                         }) {
-                        Ok(..) => debug!(
-                            target: LOG_TARGET,
-                            "Submitted block #{} to Tari node in {:.0?} (SubmitBlock)",
-                            height,
-                            start.elapsed()
-                        ),
+                        Ok(..) => {
+                            debug!(
+                                target: LOG_TARGET,
+                                "Submitted block #{} to Tari node in {:.0?} (SubmitBlock)",
+                                height,
+                                start.elapsed()
+                            );
+                            block_templates_clone.remove(&hash).await;
```

Comment:
This negates the need for the call to `block_templates_clone.remove_outdated().await;` further down completely, which I believe is actually the correct behaviour.

Reply:
This PR does not reverse that. This one is slightly different, it removes the template after a successful submit to prevent another submission for the same block which was accepted. Previously a template was removed whether the submit was successful or not (due to the difficulty being too low), which was a bug.

Reply:
I think what we have ended up doing is guarded against the same edge case in two places @hansieodendaal

Reply:
I do not agree that we should delete a template after successful submission, as it is treating a symptom and not a cause. 

As a side note, and I know this was only briefly discussed, this implementation would also not allow multiple XMRig miners connecting to the same merge mining proxy to submit competing solutions for the same block, as the first one to submit will remove the opportunity for the others to submit.

Reply:
You do make a good point.

Reply:
@hansieodendaal Why would you want to continue mining on a template that has already been accepted to the chain?

Reply:
@mikethetike we should definitely not mine on the same template again, but preventing it to be submitted to the blockchain again does not prevent mining on it.